src/test/java/ay2122s1_cs2103t_w16_2/btbb/logic/parser/util/ParserUtilTest.java

```diff
@@ -178,6 +181,52 @@ public void parseCompletionStatus_validValueWithWhitespace_returnsTrimmedComplet
         assertEquals(expectedCompletionStatus, ParserUtil.parseCompletionStatus(completionStatusWithWhitespace));
     }
 
+    @Test
+    public void parseDate_null_throwsNulPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseDate(null));
+    }
+
+    @Test
+    public void parseDate_invalidValue_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseDate(INVALID_DATE));
+    }
+
+    @Test
+    public void parseDate_validValueWithoutWhitespace_returnsLocalDate() throws Exception {
+        LocalDate expectedLocalDate = LocalDate.of(2025, 10, 21);
+        assertEquals(expectedLocalDate, ParserUtil.parseDate(VALID_DATE));
+    }
+
+    @Test
+    public void parseDate_validValueWithWhitespace_returnsLocalDate() throws Exception {
+        String dateWithWhitespace = WHITESPACE + VALID_DATE + WHITESPACE;
+        LocalDate expectedLocalDate = LocalDate.of(2025, 10, 21);
+        assertEquals(expectedLocalDate, ParserUtil.parseDate(dateWithWhitespace));
+    }
+
+    @Test
+    public void parseDates_null_throwsNulPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseDates(null));
+    }
+
+    @Test
+    public void parseDates_invalidValue_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseDates(INVALID_DATE));
+    }
+
+    @Test
+    public void parseDates_validValueWithoutWhitespace_returnsListOfLocalDates() throws Exception {
+        List<LocalDate> expectedLocalDates = List.of(LocalDate.of(2025, 10, 21));
+        assertEquals(expectedLocalDates, ParserUtil.parseDates(VALID_DATE));
+    }
+
+    @Test
+    public void parseDates_validValueWithWhitespace_returnsListOfLocalDates() throws Exception {
+        String dateWithWhitespace = WHITESPACE + VALID_DATE + WHITESPACE;
+        List<LocalDate> expectedLocalDates = List.of(LocalDate.of(2025, 10, 21));
+        assertEquals(expectedLocalDates, ParserUtil.parseDates(dateWithWhitespace));
+    }
+
```

Comment:
Add one more test case where the `parseDates` arg is actually a string with multiple dates.